app/data_providers/dispatch.py

```diff
@@ -40,15 +43,19 @@ async def get_ranked_corpus_slate(self, user_id: str, recommendation_count: int)
             logging.info(f'SetupMoment is unpersonalized for user {user_id} because no preferences were found.')
 
         items = items[:recommendation_count]
-        recommendations = [CorpusRecommendationModel(id=uuid.uuid4().hex, corpus_item=item) for item in items]
+        recommendations = [CorpusRecommendationModel(id=str(uuid.uuid4()), corpus_item=item) for item in items]
 
-        return CorpusSlateModel(
+        corpus_slate = CorpusSlateModel(
             id=self.SETUP_MOMENT_CORPUS_CANDIDATE_SET_ID,
             headline=self.DISPLAY_NAME,
             subheadline=self.SUB_HEADLINE,
             recommendations=recommendations,
         )
 
+        await self.slate_tracker.track(corpus_slate, user_id=user_id)
```

Comment:
I still think this should be done async and should not block the response since it's not critical to complete the tracking for the main functionality and it affects latency.

Reply:
I'm less concerned about latency than about additional downstream dependency which can make the service break if unavailable. We should minimize those, especially if it's a tier1 service.

Reply:
Jira task is tracked here: https://getpocket.atlassian.net/jira/software/c/projects/HS/boards/92/roadmap?selectedIssue=HS-94

Reply:
It's good that if snowplow is unavailable the method won't fail, but I'm a bit concerned about latency in this case. If, say, there is a retrying mechanism inside the library it might take significantly longer to await if snowplow is down. The general practice of graceful degradation is to make sure all the secondary functionality doesn't hurt the main one and even if something is unavailable the service still can return some results with reasonable latency. I would recommend adding todo here to explore this further.